dayrepo/snippets/master_views.py

```diff
@@ -44,4 +44,50 @@ def post(self, request: HttpRequest):
         car_trouble.save()
         return redirect(to='car_list')
 
-new_car = CarView.as_view()
\ No newline at end of file
+new_car = CarView.as_view()
+def serial_number_divide(car: Car) -> list:
+    str_number = str(car.serial_number)
+    if len(str_number) > 2:
+        border = len(str_number)-2
+        serial_number_top = str_number[:border]
+        serial_number_end = str_number[border:]
+    else:
+        serial_number_top = ""
+        serial_number_end = str_number
+    return [serial_number_top,serial_number_end]
+
+class CarEditView(View):
+    def get(self, request: HttpRequest, car_id: int):
+        car = get_object_or_404(Car, pk=car_id)
+        serial_numbers = serial_number_divide(car)
+        serial_number_top = serial_numbers[0]
+        serial_number_end = serial_numbers[1]
+        return render(
+            request, 
+            "car_edit.html", 
+            {
+                "car":car, 
+                "top":serial_number_top,
+                "end":serial_number_end,
+            }
+        )
+        
+    def post(self, request: HttpRequest, car_id: int):
+        req = request.POST
+        number = int(
+            str(req.get("serial_number_top"))
+            + str(req.get("serial_number_end"))
```

Comment:
すでに str であることがわかったので、特にキャストする必要性なさそう
```suggestion
            req.get("serial_number_top")
            + req.get("serial_number_end")
```

Reply:
HTMLが自動で変換していることなので、説明として残したいです。